hawkbit-extension-artifact-repository-s3/src/main/java/org/eclipse/hawkbit/artifact/repository/S3Repository.java

```diff
@@ -173,6 +173,8 @@ public void deleteBySha1(final String tenant, final String sha1Hash) {
         final String key = objectKey(tenant, sha1Hash);
 
         LOG.info("Deleting S3 object from bucket {} and key {}", s3Properties.getBucketName(), key);
+        final S3Object s3Object = amazonS3.getObject(s3Properties.getBucketName(), key);
```

Comment:
Why you need to retrieve the `S3Object` here? I think it is sufficient to use the `bucketName` and the key to just delete it without retrieving it before and close it immediataly, right?

Reply:
Just to make sure I understand you, you prefer a pattern like:

Reply:
Just to make sure I understand, you prefer a pattern like:
```
closeS3Object(amazonS3.getObject(s3Properties.getBucketName(), key));
```
Correct? Assigning to an intermediate variable is a pattern I started using to make debugging easier. As a former compiler writer, any compiler with simple flow analysis should make this just as efficient. But I'm willing to change the code.